### PR TITLE
Remove some tests from analyzer_tech_debt

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit b1d8521a5bff044e2fb53eb548c9810723a892e9 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61603/b1d8521a5bff044e2fb53eb548c9810723a892e9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by novikd — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/01925_join_materialized_columns.sql [hunk 1]
```diff
@@ -51,5 +52,6 @@ SELECT t1.time as talias FROM t1 JOIN t2 ON talias = t2.time;
 SELECT t1.time as talias FROM t1 JOIN t2 ON talias = t2.time_alias;
 SELECT t2.time as talias FROM t1 JOIN t2 ON t1.time = talias;
 SELECT t2.time as talias FROM t1 JOIN t2 ON t1.time_alias = talias;
-SELECT time as talias FROM t1 JOIN t2 ON t1.time = talias; -- { serverError AMBIGUOUS_COLUMN_NAME }
-SELECT time as talias FROM t1 JOIN t2 ON talias = t2.time; -- { serverError AMBIGUOUS_COLUMN_NAME }
+SELECT time as talias FROM t1 JOIN t2 ON t1.time = talias; -- { serverError AMBIGUOUS_COLUMN_NAME, INVALID_JOIN_ON_EXPRESSION }
+SELECT time as talias FROM t1 JOIN t2 ON talias = t2.time settings allow_experimental_analyzer=0; -- { serverError AMBIGUOUS_COLUMN_NAME }
```
Comment (vdimir) [inline tests/queries/0_stateless/01925_join_materialized_columns.sql:56]: I checked it once more and now I see that the error is about `time` column, but not identifier `talias`. We resolve alias defined in SELECT and used in ON section, so should also support it in analyzer (same for using https://github.com/ClickHouse/ClickHouse/pull/60182 will work only with the setting, though).

But as for the `time` identifier we resolve it from left table due to `single_join_prefer_left_table` is set by default.

File: tests/analyzer_tech_debt.txt [hunk 2]
```diff
@@ -1,11 +1,8 @@
 00223_shard_distributed_aggregation_memory_efficient
 00725_memory_tracking
-01083_expressions_in_engine_arguments
 01155_rename_move_materialized_view
 01624_soft_constraints
-01747_join_view_filter_dictionary
```
Comment (vdimir) [inline tests/analyzer_tech_debt.txt:6]: Just for a note, was fixed in #59483

**Comment by KochetovNicolai** [comment]
Merging, cause no code changes, and tests with analyzer were green in the last commit

**Pull request merged**